DOC: Moved VQ example & target from duplicate array broadcasting doc

Mukulikaa opened the pull request:
Closes #19136 

According to the plan discussed in the issue, I have added the vector quantization example to the main `Broadcasting` doc. I have tried to add some explanations to it from the [EricsBroadcastingDoc](http://scipy.github.io/old-wiki/pages/EricsBroadcastingDoc) (the original source of both the broadcasting docs). 

In an attempt to accommodate references to the orphan doc, I have moved the sphinx cross-linking target to the main Broadcasting doc. I tested it locally and it works, but I'm not sure if it's best practice to do that. 

I wanted some feedback on the crosslinking hack and whether the example reads like a tutorial. Thanks!
cc: @melissawm @rossbar

Mukulikaa committed:
DOC: Moved VQ example & target from duplicate doc

melissawm commented on doc/source/user/basics.broadcasting.rst:
```diff
@@ -184,4 +177,75 @@ Here the ``newaxis`` index operator inserts a new axis into ``a``,
+
+.. figure:: theory.broadcast_5.png
+    :alt: vector quantitization example
+    :name: figure-1
```
This is the reason for the CI failure: this image name is referenced twice (here and in the other document). Can you change this "name" argument to something different?

Mukulikaa replied:
Thanks! I was aware of that but decided to ignore it for now because eventually the other doc would be removed and it wouldn't clash anymore.  But now that I think of it, I should use a different naming system e.g. "doc-name.figure no." so that it doesn't cause a problem later on when we include figures in other documents too.

melissawm commented on doc/source/user/basics.broadcasting.rst:
```diff
@@ -1,22 +1,15 @@
 .. _basics.broadcasting:
+.. _array-broadcasting-in-numpy:
```
As a temporary fix, maybe? I'm not entirely sure, though. Maybe @rossbar can help us here

rossbar replied:
I don't think there's anything wrong with having multiple reference names point to the same target (at least, they both seem to work in this case).

Internally, the only document that uses the `array-broadcasting-in-numpy` reference is tutorial-svd, but of course there may be other libraries referencing the target via intersphinx. The approach you've chosen seems sensible to me :+1:

melissawm reviewed (commented):
Thanks @Mukulikaa - I agree with doing this in pieces so I'm ok with it for now. I wonder though if it would just be easier to remove the part you already included in the main broadcasting document from the orphan page, so we don't redo the work later.

rossbar reviewed (commented):
>  I wonder though if it would just be easier to remove the part you already included in the main broadcasting document from the orphan page, so we don't redo the work later.

+1 this would make it easier to keep track of what content has already been moved, especially if the full transfer will occur over multiple PRs.

Mukulikaa committed:
DOC: Changed figure name and removed some duplicate parts from orphan doc

melissawm commented:
Last (maybe nit?) suggestion: since we have moved content from this page to the `basics.broadcasting` document, might be useful to add a link to it from the `array-broadcasting-in-numpy` doc.

Mukulikaa commented:
I think it does make sense to do that for users who might land on the page through external sources. Does this mean we can never remove the `array-broadcasting-in-numpy` doc from the directory because it might break links in someone's blog or tutorial?

Mukulikaa committed:
DOC: Added link to updated doc in array-broadcasting-in-numpy doc

melissawm commented:
I think the test failure is unrelated.

Mukulikaa committed:
DOC: Added graphics for examples with alt-texts

1. Deleted older formats of the graphics which belonged to the orphan doc
2. Added SVGs for the same diagrams (with necessary corrections)
3. Added detailed alt-texts for each diagram and graph
4. Removed all content from the orphan doc since all of it has been moved

Mukulikaa commented:
Hi @MarsBarLee, I have converted all the diagrams in the doc to the SVG format and added alt-texts for them. It would be nice if you could review the [doc](https://github.com/numpy/numpy/pull/19369/commits/89f5cad8151ec8942ffd3eab9b9cdf42f144ec4b#diff-5fb92cc80657b79d2ec60b983c42afa825ca55127b215d71c3ba066cf74ee8de) once when you have time!

MarsBarLee commented:
Hi Mukulika, the alt-text and SVGs look good! Thank you for including those changes.

melissawm commented on doc/source/user/basics.broadcasting.rst:
```diff
@@ -47,11 +40,23 @@ array([ 2.,  4.,  6.])
+
+    *Figure 1*
+
+    *In the simplest example of broadcasting, the scalar ``b`` is
+    stretched to become an array of same shape as ``a`` so the shapes
+    are compatible for element-by-element multiplication.*
```
nit: The formatting here is a bit off because of the emphasis
```suggestion
    *In the simplest example of broadcasting, the scalar* ``b`` *is
    stretched to become an array of same shape as* ``a`` *so the shapes
    are compatible for element-by-element multiplication.*
```

melissawm reviewed (approved):
I think this looks really good, thanks @Mukulikaa !

Mukulikaa committed:
DOC: Fixed nested formatting

Co-authored-by: Melissa Weber Mendonça <melissawm@gmail.com>

melissawm commented:
@rossbar want to take a last look?

mattip commented:
[Here](https://20935-908607-gh.circle-artifacts.com/0/doc/build/html/user/basics.broadcasting.html) is the new page vs the [old page](https://numpy.org/doc/1.21/user/basics.broadcasting.html)

[Here](https://numpy.org/devdocs/user/theory.broadcasting.html) is the location of the old content, that page is [now empty](https://20935-908607-gh.circle-artifacts.com/0/doc/build/html/user/theory.broadcasting.html), but I guess it is good to keep it around for a few releases since other sites may be referencing it.

mattip commented:
Thanks @Mukulikaa
